code/_onclick/hud/meter/health.dm

```diff
@@ -28,7 +28,8 @@
 /obj/screen/meter/health/set_mob(var/mob/living/newmob)
 	.=..()
 	GLOB.updatehealth_event.register(L, src, /obj/screen/meter/proc/update)
-	L.hud_used.hud_healthbar = src
```

Comment:
this isn't a fix

The healthbar is part of the hud now, you've got to either make sure this called after the hud is set up. That var needs to be populated